v2/robotmk/src/bin/agent.rs

```diff
@@ -0,0 +1,139 @@
+use camino::Utf8PathBuf;
+use clap::Parser;
+use serde::{Deserialize, Serialize};
+use std::env::{var, VarError};
+use std::fs::read_to_string;
+use std::io;
+use std::path::Path;
+use walkdir::{DirEntry, WalkDir};
+
+#[derive(Deserialize)]
+pub struct Config {
+    pub results_directory: Utf8PathBuf,
+}
+
+#[derive(Parser)]
+#[command(about = "Robotmk agent plugin.")]
+struct Args {
+    /// Configuration file path.
+    #[clap(name = "CONFIG_PATH")]
+    pub config_path: Option<Utf8PathBuf>,
+}
+
+#[derive(Serialize)]
+pub struct ConfigError {
+    config_reading_error: String,
+}
+
+#[derive(Serialize)]
+pub struct ConfigFileContent {
+    config_file_content: String,
+}
+
+fn determine_config_path(arg: Option<Utf8PathBuf>) -> Result<Utf8PathBuf, String> {
```

Comment:
Personally, I would have split this into two functions:
```
fn determine_config_path(arg: Option<Utf8PathBuf>) -> Result<Utf8PathBuf, String> {
    Ok(arg.unwrap_or(config_path_from_env()?)
}

fn config_path_from_env() -> Result<Utf8PathBuf, String> {
    Ok(Utf8PathBuf::from(
        match var("MK_CONFDIR") {...}
    ).join("robotmk.json"))
}
```
But I think this is more a question of taste and not a hard requirement from my side.